File: example/src/jsMain/kotlin/de.jensklingenberg/example/johnny/board/Repl.kt
package de.jensklingenberg.example.johnny.board

import de.jensklingenberg.johnnyfivektwrapper.Board
import de.jensklingenberg.johnnyfivektwrapper.Led
import de.jensklingenberg.johnnyfivektwrapper.onReady
import de.jensklingenberg.johnnyfivektwrapper.utils.jsObject

fun Repl() {
    val board = Board()
    board.onReady {
        println("Ready event. Repl instance auto-initialized!")
        val led = Led(13)
        board.repl.inject(jsObject {
            on = { led.on() }
            off = { led.off() }
        })
    }
}

File: example/src/jsMain/kotlin/de.jensklingenberg/example/johnny/board/repl.js
//http://johnny-five.io/examples/repl/

var five = require("johnny-five");
var board = new five.Board();

board.on("ready", function() {
  console.log("Ready event. Repl instance auto-initialized!");

  var led = new five.Led(13);

  this.repl.inject({
    // Allow limited on/off control access to the
    // Led instance from the REPL.
    on: function() {
      led.on();
    },
    off: function() {
      led.off();
    }
  });
});